File: 3573/3573.go
func bestTimeToBuyAndSellStockV(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := bestTimeToBuyAndSellStockV(root.Left)
	r := bestTimeToBuyAndSellStockV(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 3573/3573.cpp
class Solution {
public:
    int bestTimeToBuyAndSellStockV(TreeNode* root) {
        if(!root) return 0;
        return 1+max(bestTimeToBuyAndSellStockV(root->left),bestTimeToBuyAndSellStockV(root->right));
    }
};